External/Plugins/ResultsPanel/Helpers/ResultsPanelHelper.cs

```diff
@@ -1,83 +1,249 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using PluginCore;
 using PluginCore.Localization;
 using PluginCore.Managers;
 using WeifenLuo.WinFormsUI.Docking;
 
 namespace ResultsPanel.Helpers
 {
-    class ResultsPanelHelper
+    internal static class ResultsPanelHelper
```

Comment:
All of the negative points you mention have a negative impact of only nanoseconds of difference even on several years old computers (there are benchmarks out there, some of them by MS), and most of them are impacts that happen only once, I would say the last point you mention is the more important one. I care more about the negative effects that static classes bring like not possible to switch one implementation with other, making it sometimes harder to refactor code, to test, of course in a class that keeps state it's impossible to create multiple instances, etc.

Since this is an internal class most of the negative points get diminished, but there are still some valid concerns.